director/director.go

```diff
@@ -334,14 +334,14 @@ func redirectToCache(ginCtx *gin.Context) {
 	// This is because the configuration of the origin/namespace should override the inclusion of "dirlisthost" for that origin.
 	// Listings is true by default so if it is ever set to false we should accept that config over the dirlisthost.
 	if namespaceAd.Caps.Listings && originAds[0].Listings {
-		if !namespaceAd.PublicRead && originAds[0].AuthURL != (url.URL{}) {
+		if !namespaceAd.Caps.PublicReads && originAds[0].AuthURL != (url.URL{}) {
```

Comment:
Looks like we already reading `Caps.PublicReads` here! So we should be abele to unset `PublicReads` for reading OSDF servers

Reply:
Same as for the previous comment, these changes happened in 7.9, so I think this would break a new advertisement (>v7.9.0) being used by an old director (<=v7.8.x).